[BUG] Erro ao publicar imagem no GitHub Container Registry e problemas com Melange/APKO no pipeline

## Descrição

Durante a execução do pipeline de Build e Distribuição de Pacotes, ocorreram as seguintes falhas:

1. **Erro ao fazer push da imagem para o GitHub Container Registry (ghcr.io):**
   - Mensagem de erro: `tag does not exist: ghcr.io/nataliagranato/giropops-senhas:1752289325`
   - O processo foi encerrado com exit code 1.
   - Possível causa: tentativa de publicar uma tag inexistente ou não criada corretamente pelo processo anterior do APKO/Melange.

2. **Problemas no fluxo Melange e APKO:**
   - As etapas relacionadas à construção e publicação de pacotes parecem não estar garantindo a criação e o push correto das imagens e tags.
   - Necessário investigar se os comandos de build e tagueamento estão sendo executados corretamente, e se há dependências de variáveis ou artefatos ausentes.

3. **Erro ao fazer upload do arquivo trivy-results.sarif:**
   - Mensagem de erro: `Error: Path does not exist: trivy-results.sarif`
   - Indica que o arquivo de resultados do Trivy não foi gerado, está sendo salvo em local diferente ou a etapa de scan está falhando.

Logs relevantes anexados à descrição para facilitar análise.

## Screenshots (se aplicável)

_Sem screenshots, apenas logs textuais no momento._

## Ambiente

- **Método de Execução**:
  - [ ] Kubernetes com manifestos
  - [ ] Kubernetes com Helm
  - [ ] Docker localmente

- **Sistema Operacional**: [preencher]
- **Navegador**: [preencher]
- **Versão da Aplicação**: [preencher]

## Logs e Saída de Erro

```
The push refers to repository [ghcr.io/nataliagranato/giropops-senhas]
tag does not exist: ghcr.io/nataliagranato/giropops-senhas:1752289325
Error: Process completed with exit code 1.
...
Error: Path does not exist: trivy-results.sarif
```

## Questões Relacionadas

_Nenhuma issue relacionada informada._

--- a/.github/workflows/ghcr.yml
+++ b/.github/workflows/ghcr.yml
@@ -72,7 +72,7 @@ jobs:
       id: generate-tag
       run: |
         # Obtém os primeiros 5 dígitos do hash do commit e a data
-        SHORT_HASH=$(git log -1 --pretty=format:%h | cut -c1-5)
+        SHORT_HASH=${GITHUB_SHA:0:5}
         TIMESTAMP=$(date +%Y%m%d%H%M%S)
         echo "tag=${SHORT_HASH}-${TIMESTAMP}" >> $GITHUB_ENV
         echo "tag=${SHORT_HASH}-${TIMESTAMP}" >> $GITHUB_OUTPUT

--- a/.github/workflows/ghcr.yml
+++ b/.github/workflows/ghcr.yml
@@ -15,6 +15,7 @@ jobs:
       actions: read
       contents: read
       security-events: write
+      packages: write
 
     steps:
     # Checkout do código
@@ -66,6 +67,16 @@ jobs:
         cd chainguard/environments/prd
         apko build apko-prod.yaml senhas senhas-prod.tar -k melange.rsa.pub --arch amd64
     
+    # Gerar nome único para a tag
+    - name: Gerar nome único para a tag
+      id: generate-tag
+      run: |
+        # Obtém os primeiros 5 dígitos do hash do commit e a data
+        SHORT_HASH=$(git log -1 --pretty=format:%h | cut -c1-5)
+        TIMESTAMP=$(date +%Y%m%d%H%M%S)
+        echo "tag=${SHORT_HASH}-${TIMESTAMP}" >> $GITHUB_ENV
+        echo "tag=${SHORT_HASH}-${TIMESTAMP}" >> $GITHUB_OUTPUT
+
     # Log in no GitHub Container Registry
     - name: Log in to GitHub Container Registry
       env:
@@ -73,10 +84,26 @@ jobs:
       run: echo $CR_PAT | docker login ghcr.io -u nataliagranato --password-stdin
   
     # Load da imagem de container e push para o GitHub Container Registry
-    - name: Load Docker image
+    - name: Load Docker image and push
       run: |
         cd chainguard/environments/prd 
         docker load < senhas-prod.tar
         docker images
-        docker tag senhas:latest-amd64 ghcr.io/nataliagranato/giropops-senhas:$(date +%s)
-        docker push ghcr.io/nataliagranato/giropops-senhas:$(date +%s)
+        docker tag senhas:latest-amd64 ghcr.io/nataliagranato/giropops-senhas:${{ steps.generate-tag.outputs.tag }}
+        docker push ghcr.io/nataliagranato/giropops-senhas:${{ steps.generate-tag.outputs.tag }}
+
+    # Scan de segurança com Aqua Security Trivy    
+    - name: Aqua Security Trivy
+      uses: aquasecurity/trivy-action@6c175e9c4083a92bbca2f9724c8a5e33bc2d97a5 # 0.30.0
+      with:
+        image-ref: ghcr.io/nataliagranato/giropops-senhas:${{ steps.generate-tag.outputs.tag }}
+        format: 'sarif'
+        severity: 'UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL'
+        output: 'trivy-results.sarif'
+    
+    # Envio dos resultados do Trivy para a aba de Segurança do GitHub
+    - name: Fazer upload dos resultados do Trivy para a aba de Segurança do GitHub
+      uses: github/codeql-action/upload-sarif@ff0a06e83cb2de871e5a09832bc6a81e7276941f # v3.28.18
+      if: always()
+      with:
+        sarif_file: 'trivy-results.sarif'

--- a/.github/workflows/ghcr.yml
+++ b/.github/workflows/ghcr.yml
@@ -80,8 +80,8 @@ jobs:
     # Log in no GitHub Container Registry
     - name: Log in to GitHub Container Registry
       env:
-          CR_PAT: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
-      run: echo $CR_PAT | docker login ghcr.io -u nataliagranato --password-stdin
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: echo $GITHUB_TOKEN | docker login ghcr.io -u nataliagranato --password-stdin
   
     # Load da imagem de container e push para o GitHub Container Registry
     - name: Load Docker image and push

